packages/cli/src/tools/installPods.ts

```diff
@@ -43,56 +44,85 @@ function runSudo(command: string): Promise<void> {
   });
 }
 
-async function installCocoaPods(loader: ora.Ora) {
-  loader.stop();
-
-  const withGem = 'Yes, with gem (may require sudo)';
-  const withHomebrew = 'Yes, with Homebrew';
+async function promptCocoaPodsInstallationQuestion() {
+  const promptQuestion = `CocoaPods ${chalk.dim.underline(
+    '(https://cocoapods.org/)',
+  )} ${chalk.reset.bold(
+    'is not installed. CocoaPods is necessary for the iOS project to run correctly. Do you want to install it?',
+  )}`;
+  const installWithGem = 'Yes, with gem (may require sudo)';
+  const installWithHomebrew = 'Yes, with Homebrew';
 
   const {shouldInstallCocoaPods} = await inquirer.prompt([
     {
       type: 'list',
       name: 'shouldInstallCocoaPods',
-      message: `CocoaPods ${chalk.dim.underline(
-        '(https://cocoapods.org/)',
-      )} ${chalk.reset.bold(
-        'is not installed. CocoaPods is necessary for the iOS project to run correctly. Do you want to install it?',
-      )}`,
-      choices: [withGem, withHomebrew],
+      message: promptQuestion,
+      choices: [installWithGem, installWithHomebrew],
     },
   ]);
 
-  switch (shouldInstallCocoaPods) {
-    case withGem:
-      const options = ['install', 'cocoapods', '--no-document'];
-      loader.start('Installing CocoaPods');
-      try {
-        // First attempt to install `cocoapods`
-        await execa('gem', options);
-      } catch (_error) {
-        // If that doesn't work then try with sudo
-        try {
-          await runSudo(`gem ${options.join(' ')}`);
-        } catch (error) {
-          loader.fail();
-          logger.error(error.stderr);
-
-          throw new Error(
-            `An error occured while trying to install CocoaPods, which is required by this template.\nPlease try again manually: sudo gem install cocoapods.\nCocoaPods documentation: ${chalk.dim.underline(
-              'https://cocoapods.org/',
-            )}`,
-          );
-        }
-      }
-      loader.succeed();
-      break;
-    case withHomebrew:
-      await brewInstall({
-        pkg: 'cocoapods',
-        label: 'Installing CocoaPods',
-        loader,
-      });
-      break;
+  const shouldInstallWithGem = shouldInstallCocoaPods === installWithGem;
+
+  return {
+    installWithGem: shouldInstallWithGem,
```

Comment:
`installWithGem` and `installWithHomebrew` are mutually exclusive.
Maybe you could have a `installMethod` property instead, which is equal to `homebrew` or `gem`?

You can even type this with TypeScript by doing this:

```typescript
type InstallMethod = 'homebrew' | 'gem';
```

Reply:
I just left it like this so it doesn't need to compare with a string here: https://github.com/react-native-community/cli/blob/8beab8cd631d3942d3e35428624e8f33664e6156/packages/cli/src/tools/installPods.ts#L101.

Thoughts on this @thymikee?

Reply:
I like the `installMethod`, let's do this

Reply:
2x1
I'll do it 😄

Reply:
Agree with @thib92 here. All strings are interned anyway, so it's just a cheap ref comparison. Alternatively, we could also just return the installer function. That way we need only check whether it's non-null.